app/models/summoner.rb

```diff
@@ -5,13 +5,25 @@ class Summoner < ActiveRecord::Base
   validates :riot_id, presence: true, uniqueness: true
   validates :name, presence: true
 
-  def self.add id
-    if Summoner.exists?(riot_id: id)
-      return Summoner.find_by(riot_id: id)
-    else
-      summoner = Riot.summoner_id(id)
-      summoner = summoner[id.to_s]
-      return Summoner.new(riot_id: summoner["id"], name: summoner["name"])
+  def self.add id_array
+    all_summoners = {}
+    split = id_array.each_slice(39).to_a
+    split.each do |array|
+      id_list = ""
+      array.each do |id|
+        id_list << id.to_s << ','
+      end
+      all_summoners.merge!(Riot.summoner_id(id_list))
     end
+    File.open('temp.json', 'w') do |f|
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.